### PR TITLE
Add as_borrowed function

### LINKED ISSUE #19: a way to 'extract' the borrowed value
In a few places I'm using code like

```
match cow of {
  Cow::Owned(o) -> // do something with the owned thingy
  Cow::Borrowed(b) -> // do something with the borrowed thingy (keeping it's lifetime)
}
```

I can't find a way to do that in `beef` Is there a chance to get  a function along the lines of:

```
fn a_good_name_for_this(self) -> Option<&'a T> {
...
}
```

### PR DESCRIPTION
**Pull request by Licenser**
The function will consume the Cow and return the borrowed data with the associated lifetime. It panics if the data wasn't borrowed.

closes #19

### CODE REVIEW

**Comment by CAD97** [comment]
This operation would probably be best called `unwrap_borrowed` as currently written.

If given a choice, I'd expect:

```rust
    fn as_borrowed(&self) -> Option<&'a T> {
        match self {
            Cow::Owned(_) => None,
            Cow::Borrowed(borrowed) => Some(borrowed),
        }
    }
    
    fn as_owned(&self) -> Option<&T::Owned> {
        match self {
            Cow::Owned(owned) => Some(owned),
            Cow::Borrowed(_) => None,
        }
    }
    
    fn as_owned_mut(&mut self) -> Option<&mut T::Owned> {
        match self {
            Cow::Owned(owned) => Some(owned),
            Cow::Borrowed(_) => None,
        }
    }
```

**Comment by Licenser** [comment]
that's a much better name! I'll update it to `unwrap_borrowed` :)

File: src/generic.rs [hunk 1]
```diff
@@ -88,6 +98,18 @@ where
         }
     }
 
+    /// Extracts borrowed data.
+    ///
+    /// Panics: If the Beef is owned.
+    #[inline]
+    pub fn unwrap_borrowed(self) -> &'a T {
+        let cow = ManuallyDrop::new(self);
```
Comment (maciejhirsz) [inline src/generic.rs]: Added tests in #21, this looked like a leak to me, Miri agreed :).

**Comment by maciejhirsz** [comment]
Thanks for the PR! Released as 0.4.1.

File: src/generic.rs (hunk 1 above)
Comment (Licenser) [reply src/generic.rs]: Sorry about that!

**Pull request merged**